src/java/org/pantsbuild/tools/junit/impl/ScalaTestUtil.java

```diff
@@ -5,6 +5,12 @@
 public final class ScalaTestUtil {
   private ScalaTestUtil() {}
 
+  /**
+   * org.scalatest.Suite.class, loaded with runtime reflection to avoid the
+   * scalatest dependency.
+   */
+  private static Class<?> suiteClass = null;
```

Comment:
Do you still see the slowdown if you load this eagerly in a static block?
```
private static Class<?> suiteClass = null;
static {
  suiteClass = Class.forName("org.scalatest.Suite", true, clazz.getClassLoader());
};
```

Reply:
This uses the `ClassLoader` from the test instead of the runner, so it needs the reference to `clazz` from inside the method. The reasoning behind this isn't totally clear to me but I didn't want to change any behavior.

Reply:
There is only one classloader in play, afaik: we shade the testrunner and then directly merge the classpaths.

Reply:
Makes sense to me. I made the change.